client/src/featureform/register.py

```diff
@@ -4561,23 +4569,24 @@ def register(self):
         )
 
     def features_and_labels(self) -> Tuple[List[ColumnMapping], List[ColumnMapping]]:
-        resources = [
-            {
-                "name": self.name,
-                "variant": self.variant,
-                "column": self.source_column,
-                "type": self.type,
-                "description": self.description,
-                "tags": self.tags,
-                "properties": self.properties,
-            }
-        ]
+        resource = {
+            "name": self.name,
+            "variant": self.variant,
+            "column": self.source_column,
+            "type": self.type,
+            "description": self.description,
+            "tags": self.tags,
+            "properties": self.properties,
+        }
+
         if self.resource_type == "feature":
-            features = resources
+            resource["dims"] = self.dims
```

Comment:
So the trick to get this dims / is_embedding out of this base class is to override features_and_labels in the EmbeddingColumnResourceClass.

so then you'd have something like:

def features_and_labels(self) ...
  features, labels = super().features_and_labels()
  features[0]["dims"] = self.dims
  features[0]["is_embedding"] = self.is_embedding
  return features, labels